labman/db/process.py

```diff
@@ -3563,6 +3563,13 @@ def generate_prep_information(self):
                 'RUNID']
             df = df[order]
             sio = StringIO()
+
+            # remove Study ID prepended to each entry in column 'Orig_name'
+            # before writing out to disk.
+            df['Orig_name'].replace(regex=True,
+                                    inplace=True,
+                                    to_replace=r'^\d+\.',
+                                    value=r'')
             df.to_csv(sio, sep='\t', index_label='sample_name')
```

Comment:
Is there anywhere in qiita that the sample name exists WITHOUT the study id prepended to it?  The orig_name we are working with comes from the following SQL:

`study_sample.sample_id AS orig_name`

It would be much preferable to pull the info we want from the db (if it exists there) rather than use a heuristic to cut it off afterwards!

Reply:
No, it doesn't - we overwrite the sample names. However, I have seen users that add a column with that information, which is basically a duplicated of sample_name/sample_id.

Reply:
@AmandaBirmingham I agree in principle, but it is as @antgonza says; I can say though that the study_id.sample_id pattern is very durable, and I confirmed this pattern held through across the unit-test study. I can also confirm it with all rows in study_sample, if you'd like.

Reply:
Ok, but we definitely DO have the study id somewhere in the qiita information. We are literally pulling it in the same sql query that gets orig_name:

            `SELECT`
               ` study.study_id, study_sample.sample_id,`

It seems to me that we therefore should not NEED to use a regex (which could break silently if the pattern changes, although I acknowledge that probably it won't) when we could just replace the known study_id (+ a dot) within the sample_id with "" ...

Reply:
A fair point. I can put the change in after our meeting. I suppose it's worth mentioning that in either case, the result would be no change if the assumption doesn't hold. I bring it up because my assumption is that Gail or someone would notice those right away, and as we wouldn't know how to handle this case, it's a proper mode of failure.

Reply:
@AmandaBirmingham After some additional time and effort spent on this issue, it seems as though a solution using study.study_id is not workable, as there are some cases where study_id returns, and is expected to return NULL (See line 3497). Reverting back to the previous solution, which I believe is suitable.

Reply:
If `study_id` is `None`, then that means Qiita does not have the sample metadata. If we do not have sample metadata, then we cannot plate the sample. Right?

Reply:
I believe in this case, the issue is that if study_id is NULL, and sample_id is NULL, then it should be understood that this row is control data. After working with this code a bit, I think the best thing is for it to be rewritten at some point. The same method is being used to generate output for two different files (control and samples), and conditionally modifying orig_name values has been difficult without breaking other behavior. Also, the test data is stored as a string representation of two TSV files, one for control and another for samples; this is pretty hard to work with. There's other ways of testing for equality.

Currently, I have modified the Orig_name column in the samples data and removed the leading study id from each row ('1.'). I haven't modified the test control file data, but I have confirmed its Orig_name column doesn't have study IDs prepending it.

I believe what we can expect is that sample data will have an integer value prepending the other part of the string value, separated by a dot. The previous regex expression will remove those, while leaving the orig_ids that don't match alone. I think this is good enough for now.